monitor/script.js

```diff
@@ -0,0 +1,246 @@
+          ${name
+        .split("-")
+        .join(
+          "_"
+        )}: indexingStatusForCurrentVersion(subgraphName: "messari/${name}") {
+            subgraph
+            node
+            synced
+            health
+            fatalError {
+              message
+              handler
+            }
+            nonFatalErrors {
+              message
+              handler
+            }
+            chains {
+              network
+              chainHeadBlock {
+                number
+              }
+              earliestBlock {
+                number
+              }
+              latestBlock {
+                number
+              }
+              lastHealthyBlock {
+                number
+              }
+            }
+            entityCount
+          }
+      `;
+    fullPendingQueryStr[fullPendingQueryStr.length - 1] += `        
+        ${name
+        .split("-")
+        .join(
+          "_"
+        )}_pending: indexingStatusForPendingVersion(subgraphName: "messari/${name}") {
+            subgraph
+            node
+            synced
+            health
+            fatalError {
+            message
+            handler
+            }
+            nonFatalErrors {
+            message
+            handler
+            }
+            chains {
+            network
+            chainHeadBlock {
+                number
+            }
+            earliestBlock {
+                number
+            }
+            latestBlock {
+                number
+            }
+            lastHealthyBlock {
+                number
+            }
+            }
+            entityCount
+        }
+    `;
+    if (fullCurrentQueryStr[fullCurrentQueryStr.length - 1].length > 80000) {
+      fullCurrentQueryStr[fullCurrentQueryStr.length - 1] += "}";
+      fullCurrentQueryStr.push(" query Status {");
+    }
+    if (fullPendingQueryStr[fullPendingQueryStr.length - 1].length > 80000) {
+      fullPendingQueryStr[fullPendingQueryStr.length - 1] += "}";
+      fullPendingQueryStr.push(" query Status {");
+    }
+  });
+  fullCurrentQueryStr[fullCurrentQueryStr.length - 1] += "}";
+  fullPendingQueryStr[fullPendingQueryStr.length - 1] += "}";
+
+  const currentStateDepos = deployments;
+  Object.keys(currentStateDepos).forEach((name) => {
+    deployments[name + "-pending"] = { ...deployments[name] };
+  });
+
+  const indexingStatusQueries = [
+    ...fullCurrentQueryStr,
+    ...fullPendingQueryStr,
+  ].map((query) =>
+    axios.post("https://api.thegraph.com/index-node/graphql", { query: query })
+  );
+  let indexData = [];
+  await Promise.all(indexingStatusQueries)
+    .then(
+      (response) =>
+      (indexData = response.map(
+        (resultData) => ({ data: resultData.data.data }.data)
+      ))
+    )
+    .catch((err) => console.log(err));
+  let tempCombinedIndexData = {};
+  indexData.forEach((x) => {
+    tempCombinedIndexData = { ...tempCombinedIndexData, ...x };
+  });
+  indexData = tempCombinedIndexData;
```

Comment:
doesn't `tempCombinedIndexData` end up as a copy of `indexData` or am i missing something?

Reply:
indexData before this loop is an array with two elements; an object of results from the current deployments query and an object of results from the pending deployments query. This loop merges the objects into one and saves it to indexData. Now in dev branch, I condensed this into one line with a spread operator.